source/slang/slang-emit-c-like.cpp

```diff
@@ -1111,7 +1113,20 @@ String CLikeSourceEmitter::getName(IRInst* inst)
     String name;
     if(!m_mapInstToName.tryGetValue(inst, name))
     {
-        name = generateName(inst);
+        // unmangle names, when emitting header
+        if (shouldEmitOnlyHeader())
```

Comment:
I think we should bottleneck name generating logic through the same branch regardless of `shouldEmitOnlyHeader` setting so our cpp code can actually match the header code. Is there a reason why we need different behavior then just calling `generateName` here?